integration/spark2/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala

```diff
@@ -640,7 +640,16 @@ object CarbonDataRDDFactory {
           val nodes = DistributionUtil.ensureExecutorsByNumberAndGetNodeList(nodeNumOfData,
             sqlContext.sparkContext)
           val newRdd = new DataLoadCoalescedRDD[Row](rdd, nodes.toArray.distinct)
-
+          breakable{
+            if (!FileUtils
+              .validateTableExists(carbonTable.getAbsoluteTableIdentifier.getStorePath +
+                                   CarbonCommonConstants.FILE_SEPARATOR +
+                                   "MetaData" + CarbonCommonConstants.FILE_SEPARATOR +
+                                   carbonTable.getCarbonTableIdentifier.getTableId)) {
+              LOGGER.error(s"Table does not exists: ${carbonTable.getTableUniqueName}")
+              break()
+            }
+          }
```

Comment:
This check should be done only at the final commit point of an operation. Like for data load this should be done just before writing the table status file and after global dictionary generation.